### ISSUE TITLE
Support ghc-9.8.1

### ISSUE THREAD

**Issue by erikd** [well-typed/cborg#327]
If I `allow-newer: cborg:*` then it builds with `ghc-9.8.1` . That means the new compiler can be supported with the following dependency updates:
```
> (cd serialise/ && cabal outdated)
Outdated dependencies:
base >=4.11 && <4.19 (latest: 4.19.0.0)
bytestring >=0.10.4 && <0.12 (latest: 0.12.0.2)
containers >=0.5 && <0.7 (latest: 0.7)
ghc-prim >=0.3.1.0 && <0.11 (latest: 0.11.0)
primitive >=0.5 && <0.9 (latest: 0.9.0.0)
text >=1.1 && <2.1 (latest: 2.1)
base >=4.11 && <4.19 (latest: 4.19.0.0)
bytestring >=0.10.4 && <0.12 (latest: 0.12.0.2)
containers >=0.5 && <0.7 (latest: 0.7)
primitive >=0.5 && <0.9 (latest: 0.9.0.0)
tasty >=0.11 && <1.5 (latest: 1.5)
text >=1.1 && <2.1 (latest: 2.1)
base >=4.11 && <4.19 (latest: 4.19.0.0)
bytestring >=0.10.4 && <0.12 (latest: 0.12.0.2)
deepseq >=1.0 && <1.5 (latest: 1.5.0.0)
aeson >=0.7 && <2.2 (latest: 2.2.1.0)
base >=4.11 && <4.19 (latest: 4.19.0.0)
bytestring >=0.10.4 && <0.12 (latest: 0.12.0.2)
deepseq >=1.0 && <1.5 (latest: 1.5.0.0)
ghc-prim >=0.3.1.0 && <0.11 (latest: 0.11.0)
aeson >=0.7 && <2.2 (latest: 2.2.1.0)
base >=4.11 && <4.19 (latest: 4.19.0.0)
bytestring >=0.10.4 && <0.12 (latest: 0.12.0.2)
containers >=0.5 && <0.7 (latest: 0.7)
deepseq >=1.0 && <1.5 (latest: 1.5.0.0)
ghc-prim >=0.3.1.0 && <0.11 (latest: 0.11.0)
text >=1.1 && <2.1 (latest: 2.1)

> (cd cborg-json/ && cabal outdated)
Outdated dependencies:
aeson >=0.7 && <2.2 (latest: 2.2.1.0)
base >=4.11 && <4.19 (latest: 4.19.0.0)
text >=1.1 && <2.1 (latest: 2.1)
aeson >=0.7 && <2.2 (latest: 2.2.1.0)
base >=4.7 && <4.19 (latest: 4.19.0.0)
bytestring >=0.10.4 && <0.12 (latest: 0.12.0.2)
tasty >=0.11 && <1.5 (latest: 1.5)
text >=1.1 && <2.1 (latest: 2.1)
base >=4.11 && <4.19 (latest: 4.19.0.0)
bytestring >=0.10.4 && <0.12 (latest: 0.12.0.2)
deepseq >=1.0 && <1.5 (latest: 1.5.0.0)

> (cd cbor-tool/ && cabal outdated)
Outdated dependencies:
aeson >=0.7 && <2.2 (latest: 2.2.1.0)
base >=4.11 && <4.19 (latest: 4.19.0.0)
bytestring >=0.10 && <0.12 (latest: 0.12.0.2)
text >=1.1 && <2.1 (latest: 2.1)

> (cd binary-serialise-cbor/ && cabal outdated)
Outdated dependencies:
base >=4.11 && <4.19 (latest: 4.19.0.0)

> (cd cborg && cabal outdated)
Outdated dependencies:
base >=4.11 && <4.19 (latest: 4.19.0.0)
bytestring >=0.10.4 && <0.12 (latest: 0.12.0.2)
containers >=0.5 && <0.7 (latest: 0.7)
deepseq >=1.0 && <1.5 (latest: 1.5.0.0)
ghc-prim >=0.3.1.0 && <0.11 (latest: 0.11.0)
primitive >=0.5 && <0.9 (latest: 0.9.0.0)
text >=1.1 && <1.3 || >=2.0 && <2.1 (latest: 2.1)
aeson >=0.7 && <2.2 (latest: 2.2.1.0)
base >=4.11 && <4.19 (latest: 4.19.0.0)
bytestring >=0.10.4 && <0.12 (latest: 0.12.0.2)
deepseq >=1.0 && <1.5 (latest: 1.5.0.0)
primitive >=0.5 && <0.9 (latest: 0.9.0.0)
tasty >=0.11 && <1.5 (latest: 1.5)
text >=1.1 && <2.1 (latest: 2.1)
```

These can all be done as a Hackage metadata edit.

Let me know if you would prefer a PR.

**Comment by phadej** [comment]
Adding support for `primitive-0.9` would be nice too. (Though `primitive-0.8` works with GHC-9.8 too)

**Comment by erikd** [comment]
PR is https://github.com/well-typed/cborg/pull/328

**Comment by andreasabel** [comment]
The linked PR also fixes the `aeson < 2.2` problem which risks `cborg` to drop out of Stackage:
- https://github.com/commercialhaskell/stackage/issues/7031

ATTN: @Saizan (Hello Andrea! 🎉 )

**Comment by adamgundry** [comment]
Looks like this was resolved by https://github.com/well-typed/cborg/pull/328#pullrequestreview-1728529400
